kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Context.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.service;
+
+import io.kroxylicious.proxy.config.BaseConfig;
+
+/**
+ * Context in which a Contributor is getting an instance, this includes the user-supplied configuration.
+ */
+public interface Context {
+
+    /**
+     * service configuration which may be null if the service instance does not accept configuration.
+     * @return config
+     */
+    BaseConfig getConfig();
```

Comment:
Maybe not part of this PR.

Can we add something like `<C extends BaseConfig> C getMyConfig(TypeReference<C> tr)` ? and does it help provide type safe config where we need it?

Reply:
Definitely keen to defer any parameterization of BaseConfig, this is where the wheels have come off a couple of times while I've refactored, so if it can be done in it's own small PR that'd help

Reply:
Yeah, ok lets review that one later.